src/website/webview/views.py

```diff
@@ -490,22 +490,57 @@ def get_context_data(self, **kwargs: Any) -> Any:
 
         # Major channels are the important channels that a user wants to keep an eye on.
         # FIXME make it dynamic
-        major_channels = ["nixos-23.11", "nixos-24.05", "nixos-24.11", "nixos-unstable"]
+        major_channels = ["23.11", "24.05", "24.11", "unstable"]
+
+        def get_major_channel(branch_name: str) -> str | None:
+            for mc in major_channels:
+                if mc in branch_name:
+                    return f"nixos-{mc}"
+            return None
 
         for obj in context["objects"]:
+            # TODO Validate assumption that there is only one container
             obj.cve_container = obj.cve.container_set.all().first()
             obj.packages = dict()
             for derivation in obj.derivations.all():
                 attribute = derivation.attribute.removesuffix(f".{derivation.system}")
-                channel = derivation.parent_evaluation.channel.channel_branch
+                branch_name = derivation.parent_evaluation.channel.channel_branch
                 # FIXME This is wrong. Replace with something like builtins.parseDrvName
                 version = derivation.name.split("-")[-1]
+                major_channel = get_major_channel(branch_name)
                 if attribute not in obj.packages:
                     obj.packages[attribute] = dict()
-                obj.packages[attribute][channel] = {
-                    "version": version,
-                    "major": channel in major_channels,
-                }
+                # FIXME This quietly drops unfamiliar branch names
+                if major_channel:
+                    if major_channel not in obj.packages[attribute]:
```

Comment:
Can the channel analyzing logic be moved in its own functions and possibly a utility file for channel analysis?

Reply:
Definetly, this entire logic is way too noisy for the view.

Reply:
This is probably wrong but I ended up moving this into the models for now? I thought it might make sense, as the function operates on the types in the `nix_evaluation` model.